cloud-sql.mk

```diff
@@ -7,24 +7,27 @@ install-cloud-sql-proxy:
 		curl -o cloud_sql_proxy https://dl.google.com/cloudsql/cloud_sql_proxy.darwin.amd64 && \
 		chmod +x cloud_sql_proxy
 
-test-cloud-sql-config:
+batch-secrets/batch-test-cloud-sql-config:
 	mkdir -p batch-secrets && kubectl get secret \
 	    batch-test-cloud-sql-config -n batch-pods \
 	    -o "jsonpath={.data.batch-test-cloud-sql-config\.json}" \
 	  | base64 --decode \
 	  | jq '.host = "127.0.0.1" | .port = $(CLOUD_SQL_PORT)' \
 	    > batch-secrets/batch-test-cloud-sql-config.json
 
-production-cloud-sql-config:
+batch-secrets/batch-production-cloud-sql-config:
 	mkdir -p batch-secrets && kubectl get secret \
 	    batch-production-cloud-sql-config -n default \
 	    -o "jsonpath={.data.batch-production-cloud-sql-config\.json}" \
 	  | base64 --decode \
 	  | jq '.host = "127.0.0.1" | .port = $(CLOUD_SQL_PORT)' \
 	    > batch-secrets/batch-production-cloud-sql-config.json
```

Comment:
This is probably better represented as $@

Reply:
Can you be more specific? I'm still learning how to make Makefiles.

Reply:
Oh, $@ represents the target name. So if here it will represent the target name:

`batch-secrets/batch-production-cloud-sql-config`

I suggested that you change the target name to 

`batch-secrets/batch-production-cloud-sql-config.json`. Then, this target name will be substituted anywhere "$@" is specified in the body of the rule, so you don't need to type filename twice. Since the target name is now a file on the filesystem (provided it isn't specified in .PHONY), this rule will avoid re-making when the target (file) exists. This use of the filesystem is idiomatic use of make afaik.

Reply:
https://stackoverflow.com/questions/3220277/what-do-the-makefile-symbols-and-mean

For example, consider the following declaration:
```make
all: library.cpp main.cpp
```

In this case:

`$@` evaluates to all
`$<` evaluates to library.cpp
`$^` evaluates to library.cpp main.cpp